internal/component/loki/source/file/internal/tail/signature.go

```diff
@@ -0,0 +1,77 @@
+package tail
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"io"
+	"os"
+)
+
+// signature represents the first N bytes of a file, used to detect atomic writes
+// where a file is replaced but may have the same initial content.
+type signature struct {
+	d []byte
+}
+
+// signatureSize is the target size for a complete signature.
+const signatureSize = 512
+
+// signatureThresholds defines the byte offsets at which we should recompute the signature
+// as the file grows. This allows us to progressively build a more complete signature.
+var signatureThresholds = []int{64, 128, 256, 512}
```

Comment:
Why add this complication? could we wait for the signatureSize and assume it's 0 at first? or unknown?